### ISSUE TITLE
Web: desktop-reserved keybindings non functional in fullscreen Firefox

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#83343]
Maybe we cannot get this to work, but: when I use Firefox and go fullscreen, hitting a key such as `Cmd+W` to close an editor is still closing the entire window. That makes it almost impossible to use VSCode in Firefox.

To make matters worse: simply pressing `Esc`-key gets out of fullscreen instantly....

**Comment by bpasero** [comment]
I think we should investigate for November to see what is missing here. 

@rebornix where is the code that checks for being able to bind all our commands?

**Comment by rebornix** [comment]
At this moment we only bring back all keybindings when `navigator.keyboard` exits. But based on my current testing, it seems if in Full Screen Mode, we can actually receive most of the keyboard events.

https://github.com/microsoft/vscode/blob/7ab195a8973c74871c22b2add0f810b716028940/src/vs/workbench/services/keybinding/browser/keybindingService.ts#L245-L249 We can relax it by bringing back all keybindings when it's full screen. It's easy change but I'll do that once I tested against Safari and Firefox.

**Comment by bpasero** [comment]
> At least for now, I didn't see a plain simple solution for Firefox.

Yeah, Firefox seems really hard to get working. No matter what, in fullscreen a simple ESC press exits fullscreen...

> I'm also wondering whether it's because I upgraded to Catalina as previously, Keybindings in Full Screen in Safari doesn't seem to work. I need to double check.

If I disable the code in `keybindings.ts` that requires navigator.keyboard, I can get e.g. `Cmd+N` to work in Safari when I am in fullscreen mode. I am on macOS 10.14.6. Maybe we should be less strict and not require `navigator.keyboard` for our logic?

**Comment by rebornix** [comment]
> Maybe we should be less strict and not require navigator.keyboard for our logic?

I think so, I relaxed it a bit in #83957. For Firefox, there are two issues we can track

* keyboard.lock https://bugzilla.mozilla.org/show_bug.cgi?id=700123, it's labeled as P3
* keyboard shortcuts in full screen https://bugzilla.mozilla.org/show_bug.cgi?id=1593883, which I opened yesterday and still wait for triage.

**Comment by djbrown** [comment]
backreference to closed unanswered https://github.com/orgs/community/discussions/40566
